### PR TITLE
fix: Collection page query over-fetching

### LINKED ISSUE #11038: Collection page query over-fetching
that request took some time , `ahk/collection/u-2024` has 12k nfts

do we need all those nfts to be fetched ? 

CleanShot 2024-09-24 at 09 38 43@2x

### LINKED ISSUE #11041: Non Drop Collection showing `Explore Algorithm`
### What happened?

CleanShot 2024-09-25 at 09 01 49@2x

### LINKED ISSUE #11046: normal nft collection view having explore algorithm
image

this button should not be there for non-gen art collections 

https://canary.kodadot.xyz/ahp/collection/238

### LINKED ISSUE #11055: Collection activity chart not loading
The chart can't be shown. It could be fixed on another issue as it's also happening in canary on my side.

image

image

_Originally posted by @Jarsen136 in https://github.com/kodadot/nft-gallery/issues/11039#issuecomment-2381356447_

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- fix: removed unnecessary extra call with all nfts
- fix: removed duplicate `get collection drops` api calls
- perf:`get collection drops` earlier
- [x] Closes #11038
- [x] Closes #11041
- [x] Closes #11046
- [x] Closes #11055

### Before

CleanShot 2024-09-25 at 11 59 01@2x

CleanShot 2024-09-25 at 11 52 19@2x

### After

CleanShot 2024-09-25 at 11 53 59@2x

CleanShot 2024-09-25 at 11 59 25@2x

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

### Before

CleanShot 2024-09-25 at 12 06 01@2x

### After

CleanShot 2024-09-25 at 12 01 13@2x

### CODE REVIEW

**Comment by Jarsen136** [comment]
Not sure if it's my network. It did not show the basic info about collection when I opened the link:
https://deploy-preview-11039--koda-canary.netlify.app/ahk/collection/u-2024
image

image

**Comment by hassnian** [comment]
@Jarsen136 fixed, can you check again

CleanShot 2024-09-28 at 09 52 46@2x

**Comment by Jarsen136** [comment]
The chart can't be shown. It could be fixed on another issue as it's also happening in Canary.

image
image

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Comment by hassnian** [comment]
@Jarsen136 added https://github.com/kodadot/nft-gallery/issues/11055 in this pr  [8a0cc9d](https://github.com/kodadot/nft-gallery/pull/11039/commits/8a0cc9d654a0453fdaeabef4382ca5ee19b3187a)

that page is heavy  so the issue might be that it takes a lot of time,, it's working on my end ,  can you try again ?


CleanShot 2024-10-02 at 12 22 17

**Comment by Jarsen136** [comment]
> @Jarsen136 added #11055 in this pr -> [8a0cc9d](https://github.com/kodadot/nft-gallery/pull/11039/commits/8a0cc9d654a0453fdaeabef4382ca5ee19b3187a)
> 
> that page is heavy so the issue might be that it takes a lot of time ( related with #10416 ) , it's working on my end , can you try again ?
> 

Awesome, it works well on my side. Thank you!
image

File: components/collection/CollectionHeader.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+<script lang="ts" setup>
+import { useCollectionMinimal } from '@/components/collection/utils/useCollectionDetails'
+import CollectionInfo from '@/components/collection/CollectionInfo.vue'
+import CollectionBanner from '@/components/collection/CollectionHeader/CollectionBanner.vue'
+
+const route = useRoute()
+const collectionId = computed(() => route.params.id.toString())
+
+const { collection, refetch } = useCollectionMinimal({
```
Comment (preschian) [inline components/collection/CollectionHeader.vue:27]: `collection` doesnt have proper types before. would be good if we can fix that here

**Review by preschian — APPROVED** [review]
nice

**Pull request merged**